src/batched/KokkosBatched_SolveLU_Serial_Impl.hpp

```diff
@@ -0,0 +1,128 @@
+#ifndef __KOKKOSBATCHED_SOLVELU_SERIAL_IMPL_HPP__
+#define __KOKKOSBATCHED_SOLVELU_SERIAL_IMPL_HPP__
+
+
+/// \author Vinh Dang (vqdang@sandia.gov)
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_Trsm_Decl.hpp"
+#include "KokkosBatched_Trsm_Serial_Impl.hpp"
+
+namespace KokkosBatched {
+  namespace Experimental {
+    ///
+    /// Serial Impl
+    /// =========
+
+    ///
+    /// SolveLU no piv
+    ///
+
+    template<>
+    template<typename AViewType,
+             typename BViewType>
+    KOKKOS_INLINE_FUNCTION
+    int
+    SerialSolveLU<Algo::SolveLU::Unblocked,Trans::NoTranspose>::
+    invoke(const AViewType &A, 
+           const BViewType &B) {
+        static_assert(AViewType::rank == 2, "A should have two dimensions");
+        static_assert((BViewType::rank == 1)||(BViewType::rank == 2), "B should have either one dimension or two dimensions");
+        static_assert(std::is_same<typename AViewType::memory_space, typename BViewType::memory_space>::value, "A and B should be on the same memory space");
+        assert(A.extent(0)==A.extent(1));
+        assert(A.extent(1)==B.extent(0));
+
+        typedef typename AViewType::value_type ScalarType;
+
+        const ScalarType one(1.0);
+        
+        //First, compute Y (= U*X) by solving the system L*Y = B for Y
+        SerialTrsm<Side::Left,Uplo::Lower,Trans::NoTranspose,Diag::Unit,Algo::Trsm::Unblocked>::invoke(one, A, B);
+        //Second, compute X by solving the system U*X = Y for X
+        SerialTrsm<Side::Left,Uplo::Upper,Trans::NoTranspose,Diag::NonUnit,Algo::Trsm::Unblocked>::invoke(one, A, B);
+        
+        return 0;
```

Comment:
Why return an error code if it's always zero?  It would be better to follow the LAPACK error code convention here, for example by returning an error code instead of `assert`ing on input dimension error.

Reply:
No error checking is intended as this is designed for small batch codes. It means that the code is used in parallel for. Unfortunately, error checking brings some amount of overhead and it is difficult (maybe impossible if the device is a GPU) to handle the error during the parallel run. So, the user should understand that the batched blas routines do not check errors and the numeric error should be checked in a higher level when the user think it is necessary.

Reply:
That's fine, but then why return an integer?  Why not just make these functions `void`?

Reply:
I reserve the integer for a potential use case if an application has a strong use case that requires error checking.